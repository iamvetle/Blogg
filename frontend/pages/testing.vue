<template>
  <div class="flex">

    <form @submit.prevent method="post" class="flex flex-col" enctype="multipart/form-data">

      <label for="name">Name:</label>
      <input type="text" name="name" id="name">

      <label for="email">Email</label>
      <input type="email" name="email" id="email">

      <label for="message">Message:</label>
      <textarea name="message" id="" cols="30" rows="10" placeholder="Write a message here"></textarea>

      <label for="file">File:</label>
      <input type="file" name="file" id="file">

      <button type="submit">Submit</button>
    </form>

  </div>
</template>

<script setup lang="ts">
onMounted(() => {

  const form = document.querySelector("form")

  form?.addEventListener("submit", (e: any) => {


    const formData = new FormData(form)

    for (let item of formData) {
      console.log(item[0], item[1])
    }
    
    // ta axios of prÃ¸v fetch her
  })





})

// watchEffect(() => {

// })

</script>

<style scoped></style>