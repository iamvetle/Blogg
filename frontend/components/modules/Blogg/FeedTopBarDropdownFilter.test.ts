import { shallowMount } from '@vue/test-utils';
import FeedTopBarDropdownFilter from '~/components/modules/Blogg/FeedTopBarDropdownFilter.vue';
// import { useSearchStore } from '~/store/searchStore';
import { createTestingPinia } from '@pinia/testing';

let wrapper:any;
let searchStore;
let pinia = createTestingPinia()
let paginationStore = usePaginationStore()

const factory = () => {
    return shallowMount(FeedTopBarDropdownFilter, {
        global: {
            plugins: [pinia],
            components: {},
            mocks: {},
            stubs: {},
        },
        props: {},
        slots: {}
    })
}

describe('Testing the FeedTopBarDropdownFilter component', () => {

    beforeEach(() => {
        searchStore = useSearchStore(pinia)
        paginationStore = usePaginationStore()
        
    })

    it('renders with the correct placeholder', () => {
        wrapper = factory()
        expect(wrapper.exists()).toBe(true)
    });
    test('Should match snapshot', () => {
        wrapper = factory()
        expect(wrapper.html()).toMatchSnapshot()
    })
    

});
