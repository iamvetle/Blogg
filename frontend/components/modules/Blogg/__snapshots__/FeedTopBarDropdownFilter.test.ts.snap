// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the FeedTopBarDropdownFilter component > Should match snapshot 1`] = `
"<div data-test=\\"filter_tags\\">
  <uselectmenu options=\\"\\" modelvalue=\\"\\" searchable=\\"\\" searchableplaceholder=\\"Search a tag...\\" multiple=\\"\\" placeholder=\\"Option tags\\" size=\\"lg\\" selectclass=\\"items-center bg-primaryContainer text-onPrimaryContainer outline-none text-sm\\"></uselectmenu>
</div>"
`;
